### PR TITLE
stm32/bsp: Fix build for sensorhub

### PR DESCRIPTION
**Pull request by kasjer**
- AHBPrescTable had different type then in includes.
- APBPrescTable was missing.
- Flash sector defintion was not updated as other BSP were moved to
  have common code base.

### CODE REVIEW

File: hw/bsp/sensorhub/src/hal_bsp.c [hunk 1]
```diff
@@ -69,7 +90,7 @@ hal_bsp_flash_dev(uint8_t id)
     if (id != 0) {
         return NULL;
     }
-    return &stm32f4_flash_dev;
```
Comment (utzig) [inline hw/bsp/sensorhub/src/hal_bsp.c]: I did not expect this to still exist, does this BSP build?

Comment (kasjer) [reply hw/bsp/sensorhub/src/hal_bsp.c]: blinky builds with this BSP and changes in this PR.
Not tested though.

Comment (utzig) [reply hw/bsp/sensorhub/src/hal_bsp.c]: A grep returns only `extern const struct hal_flash stm32f4_flash_dev;` in `hw/mcu/stm/stm32f4xx/include/mcu/stm32f4_bsp.h`, weird!

Comment (utzig) [reply hw/bsp/sensorhub/src/hal_bsp.c]: Btw, do you have any idea what board this is?

Comment (kasjer) [reply hw/bsp/sensorhub/src/hal_bsp.c]: I have no clue, I spent few minutes to find, and failed

Comment (utzig) [reply hw/bsp/sensorhub/src/hal_bsp.c]: Same here! ;-) I assume it might be some custom board contributed way back that might not be available anymore...

**Pull request merged**